app/src/main/java/com/theempire/fieldform/workers/SyncWorker.kt

```
@@ -16,15 +22,27 @@ class SyncWorker(
-                repo.update(visit.copy(status = "ERROR"))
+        val pendingAndErrorVisits = repo.getAllExceptSent()
+        if (pendingAndErrorVisits.isNotEmpty()) {
+            remoteRepo.sendVisitToServer(pendingAndErrorVisits).collect { success ->
+                if (success) {
+                    pendingAndErrorVisits.forEach { visit ->
+                        try {
+                            withContext(kotlinx.coroutines.Dispatchers.IO) {
+                                delay(500)
+                                repo.update(visit.copy(status = StatusType.SENT.toString()))
+                            }
+                        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)